### PR TITLE
[nvidia-bluefield] Add config inclusions for Nvidia Bluefield-3 DPU

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
Nvidia BF-3 Data Processing Unit is a 400 gigabits per second (Gb/s) infrastructure compute platform with
line-rate processing of software-defined networking. Nvidia BF-3 DPU can be used as a standalone appliance
as well as one of the modules in the Smart Switch device.

### CODE REVIEW

**Review by vivekrnv — APPROVED** [review]

File: patch/kconfig-secure-boot-inclusions [hunk 1]
```diff
@@ -11,7 +11,7 @@ CONFIG_MODULE_SIG_FORCE=y
 CONFIG_SYSTEM_TRUSTED_KEYS="debian/certs/debian-uefi-certs.pem"
 CONFIG_MODULE_SIG_HASH="sha512"
 CONFIG_MODULE_SIG_SHA512=y
-CONFIG_KEXEC_SIG_FORCE=y
+CONFIG_KEXEC_SIG=y
```
Comment (paulmenzel) [inline patch/kconfig-secure-boot-inclusions:14]: Please make this a separate commit.

**Review by paulmenzel — CHANGES_REQUESTED** [review]
Thank you for your patch. Some small nits:

1.  The [official spelling seems to be *BlueField*](https://www.nvidia.com/en-us/networking/products/data-processing-unit/).
2.  Please split the commit into two.
3.  Please amend the commit message to remove the trailing dot from the commit message summary/title, and to give a little more background about the system. For example, would BlueField-2 now also be supported?

File: patch/kconfig-secure-boot-inclusions (hunk 1 above)
Comment (oleksandrivantsiv) [reply patch/kconfig-secure-boot-inclusions]: I moved those changes into a separate PR: https://github.com/sonic-net/sonic-linux-kernel/pull/381

**Comment by oleksandrivantsiv** [comment]
> Thank you for your patch. Some small nits:
> 
> 1. The [official spelling seems to be _BlueField_](https://www.nvidia.com/en-us/networking/products/data-processing-unit/).
> 2. Please split the commit into two.
> 3. Please amend the commit message to remove the trailing dot from the commit message summary/title, and to give a little more background about the system. For example, would BlueField-2 now also be supported?

Thank you Paul for your feedback. 
1. Both `Bluefield` and `BlueField` are correct spellings as well as BF-3. I prefer `Bluefield`
2. The default merge strategy in the repository is "squash". Even if I split the changes into two commits they will be squashed together during PR merge. Instead, I split the changes into separate PR: https://github.com/sonic-net/sonic-linux-kernel/pull/381
3. Only BF-3 is supported. We have no plans to add BF-2 support.

File: patch/kconfig-inclusions [hunk 2]
```diff
@@ -200,4 +200,15 @@ CONFIG_I2C_MUX_GPIO=m
...
+[nvidia-bluefield]
+CONFIG_MMC_DW=y
+CONFIG_MMC_DW_PLTFM=y
+CONFIG_MMC_DW_BLUEFIELD=y
+CONFIG_ARM64=y
+CONFIG_ACPI=y
+CONFIG_I2C=y
+CONFIG_I2C_SLAVE=y
+CONFIG_ARCH_SUPPORTS_ACPI=y
+CONFIG_EFIVAR_FS=y
```
Comment (saiarcot895) [inline patch/kconfig-inclusions:212]: These lines are, I think, already set by other options or default values. Can they be removed?

Comment (vivekrnv) [reply patch/kconfig-inclusions:212]: I think none of these are common or repeated

Comment (saiarcot895) [reply patch/kconfig-inclusions:212]: They would generally apply to all ARM64 platforms. `CONFIG_ARM64`, for example, would also be needed by marvell-arm64.

Comment (vivekrnv) [reply patch/kconfig-inclusions:212]: Okay, Makes sense to move CONFIG_ARM64 into the common arm64 section. 

But for others we are not certain that they can be safely moved to common section. If you think so, we can update.

Comment (oleksandrivantsiv) [reply patch/kconfig-inclusions]: @saiarcot895 I removed all options enabled by default. Please check

**Review by saiarcot895 — APPROVED** [review]

**Comment by oleksandrivantsiv** [comment]
@saiarcot895 can we merge this PR?
